src/test/java/com/mavenit/argos/pages/ResultsPage.java

```diff
@@ -0,0 +1,86 @@
+package com.mavenit.argos.pages;
+
+import com.mavenit.argos.driver.DriverFactory;
+import com.mavenit.argos.utils.Helpers;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ResultsPage extends DriverFactory {
+
+    public String getSearchTitle() {
+        return driver.findElement(By.className("search-title__term")).getText();
+    }
+
+    public List<String> getAllProductNames() {
+        List<String> productNamesList = new ArrayList<>();
+        List<WebElement> productWebElements = isProductsAvailable();
+        for (WebElement indProduct : productWebElements) {
+            String actual = indProduct.getText();
+            productNamesList.add(actual);
+        }
+        return productNamesList;
+    }
+
+    public List<String> getAllProductRatings() {
+        List<String> productRatingList = new ArrayList<>();
+        List<WebElement> productWebElements = isProductRatingAvailable();
+        for (WebElement indProduct : productWebElements) {
+            String currentRating = indProduct.getAttribute("data-star-rating");
+            productRatingList.add(currentRating);
+        }
+        return productRatingList;
+    }
+
+    public String selectAnyProduct() {
+        List<WebElement> productWebElements = isProductsAvailable();
+        int productSize = productWebElements.size();
+
+        int randomNumber = new Helpers().randomNumberGenerator(productSize);
+
+        WebElement selectedElement = productWebElements.get(randomNumber);
+        String selectedProductName = selectedElement.getText();
+        selectedElement.click();
+
+        return selectedProductName;
+
+    }
+
+    private List<WebElement> isProductsAvailable() {
+        List<WebElement> productWebElements = driver.findElements(By.cssSelector("a[data-test='component-product-card-title']"));
+        if (productWebElements.size() == 0) {
+            // fail("Zero products found .....");
+            throw new RuntimeException("Zero products found .....");
+        }
+        return productWebElements;
+    }
+
+    private List<WebElement> isProductRatingAvailable() {
+        List<WebElement> productWebElements = driver.findElements(By.cssSelector("div[data-test=\"component-ratings\"]"));
+        if (productWebElements.size() == 0) {
+            // fail("Zero products found .....");
+            throw new RuntimeException("Zero products found .....");
```

Comment:
Duplicate method as above

Reply:
i removed this below code. technically there is no need for this code on both methods
if (productWebElements.size() == 0) {
            // fail("Zero products found .....");
            throw new RuntimeException("Zero products found .....");

 , and used only in the search term, in the begining when we search any product and if product is not available then we will get this runtimeException error.